ibis/backends/dask/executor.py

```diff
@@ -0,0 +1,414 @@
+    ):
+        """Execute an elementwise UDF."""
+
+        def mapper(df):
+            cols = [df[col] for col in df]
+            return func(*cols)
+
+        df, _ = cls.asframe(func_args)
+        result = df.map_partitions(mapper)
+        if op.dtype.is_struct():
+            result = result.apply(lambda row: row.to_dict(), axis=1)
+        return result
+
+    ############################# Reductions ##################################
+
+    @classmethod
+    def visit(cls, op: ops.ArgMin | ops.ArgMax, arg, key, where):
+        # TODO(kszucs): raise a warning about triggering compute()?
+        if isinstance(op, ops.ArgMin):
+            func = lambda x: x.idxmin()
+        else:
+            func = lambda x: x.idxmax()
+
+        if where is None:
+
+            def agg(df):
+                indices = func(df[key.name])
+                if isinstance(indices, (dd.Series, dd.core.Scalar)):
+                    # to support both aggregating within a group and globally
+                    indices = indices.compute()
+                return df[arg.name].loc[indices]
+        else:
+
+            def agg(df):
+                mask = df[where.name]
+                filtered = df[mask]
+                indices = func(filtered[key.name])
+                if isinstance(indices, (dd.Series, dd.core.Scalar)):
+                    # to support both aggregating within a group and globally
+                    indices = indices.compute()
+                return filtered[arg.name].loc[indices]
+
+        return agg
+
+    @classmethod
+    def visit(cls, op: ops.Correlation, left, right, where, how):
+        if where is None:
+
+            def agg(df):
+                return df[left.name].corr(df[right.name])
+        else:
+
+            def agg(df):
+                mask = df[where.name]
+                lhs = df[left.name][mask].compute()
+                rhs = df[right.name][mask].compute()
+                return lhs.corr(rhs)
+
+        return agg
+
+    @classmethod
+    def visit(cls, op: ops.Covariance, left, right, where, how):
+        # TODO(kszucs): raise a warning about triggering compute()?
+        ddof = {"pop": 0, "sample": 1}[how]
+        if where is None:
+
+            def agg(df):
+                lhs = df[left.name].compute()
+                rhs = df[right.name].compute()
+                return lhs.cov(rhs, ddof=ddof)
+        else:
+
+            def agg(df):
+                mask = df[where.name]
+                lhs = df[left.name][mask].compute()
+                rhs = df[right.name][mask].compute()
+                return lhs.cov(rhs, ddof=ddof)
+
+        return agg
+
+    @classmethod
+    def visit(
+        cls, op: ops.ReductionVectorizedUDF, func, func_args, input_type, return_type
+    ):
+        def agg(df):
+            # if df is a dask dataframe then we collect it to a pandas dataframe
+            # because the user-defined function expects a pandas dataframe
```

Comment:
If the UDF expects a pandas dataframe, shouldn't we let the user do the compute on their own and then pass it through? It looks to me that we don't support udfs for dask dataframes, and we are triggering a compute so it'll be a pandas df. If the dataframe is big, then this will break things, and the user might not understand why.

Reply:
I agree that computing here like this would be unexpected and not something we should do.

IIUC these are the legacy udfs. IMO we should remove these as part of this release. They were only supported by the pandas and dask backends, and given the rewrite changed a bunch of the internals (and the only major user of these we know of made _heavy_ use of the internals) I suspect there's no user that would be able to upgrade to this release that was also using the legacy udfs. We should drop them for them IMO for the new udfs alone.